File: generated-code/javascript/model/repair-properties.js

class RepairProperties {
    constructor(validTargets, power) {
        this.validTargets = validTargets;
        this.power = power;
    }

    static async readFrom(stream) {
        let validTargets;
        validTargets = [];
        for (let validTargetsCount = await stream.readInt(); validTargetsCount > 0; validTargetsCount--) {
            let validTargetsElement;
            validTargetsElement = await stream.readInt();
            validTargets.push(validTargetsElement);
        }
        let power;
        power = await stream.readInt();
        return new RepairProperties(validTargets, power)
    }

    async writeTo(stream) {
        let validTargets = this.validTargets;
        await stream.writeInt(validTargets.length);
        for (let validTargetsElement of validTargets) {
            await stream.writeInt(validTargetsElement);
        }
        let power = this.power;
        await stream.writeInt(power);
    }
}
module.exports = RepairProperties

File: generated-code/kotlin/codecraft/TcpReadWrite/src/main/kotlin/trans_gen_test/model/RepairProperties.kt
package trans_gen_test.model

import trans_gen_test.util.StreamUtil

/**
 * Entity's repair properties
 */
class RepairProperties {
    /**
     * Valid target entity types
     */
    var validTargets: Array<trans_gen_test.model.EntityType>
    /**
     * Health restored in one tick
     */
    var power: Int

    constructor(validTargets: Array<trans_gen_test.model.EntityType>, power: Int) {
        this.validTargets = validTargets
        this.power = power
    }

    /**
     * Write RepairProperties to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, validTargets.size)
        for (validTargetsElement in validTargets) {
            StreamUtil.writeInt(stream, validTargetsElement.tag)
        }
        StreamUtil.writeInt(stream, power)
    }

    /**
     * Get string representation of RepairProperties
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("RepairProperties { ")
        stringBuilder.append("validTargets: ")
        stringBuilder.append("[ ")
        var validTargetsIndex = 0
        for (validTargetsElement in validTargets) {
            if (validTargetsIndex != 0) {
                stringBuilder.append(", ")
            }
            stringBuilder.append(validTargetsElement)
            validTargetsIndex++
        }
        stringBuilder.append(" ]")
        stringBuilder.append(", ")
        stringBuilder.append("power: ")
        stringBuilder.append(power)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read RepairProperties from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): RepairProperties {
            var validTargets: Array<trans_gen_test.model.EntityType>
            validTargets = Array(StreamUtil.readInt(stream), {
                var validTargetsElement: trans_gen_test.model.EntityType
                validTargetsElement = trans_gen_test.model.EntityType.readFrom(stream)
                validTargetsElement
            })
            var power: Int
            power = StreamUtil.readInt(stream)
            return RepairProperties(validTargets, power)
        }
    }
}